contracts/libs/gov-pool/GovPoolRewards.sol

```diff
@@ -33,6 +34,12 @@ library GovPoolRewards {
 
         uint256 rewards = pendingRewards[proposalId][msg.sender];
 
+        address nftMultiplier = IGovPool(address(this)).nftMultiplier();
```

Comment:
Let's move the rewards multiplier from this function to the `updateRewards` function, I see several attack vectors with the current logic

Reply:
done